modules/axtask/src/run_queue.rs

```diff
@@ -84,49 +257,56 @@ impl AxRunQueue {
     pub fn exit_current(&mut self, exit_code: i32) -> ! {
         let curr = crate::current();
         debug!("task exit: {}, exit_code={}", curr.id_name(), exit_code);
-        assert!(curr.is_running());
+        assert!(curr.is_running(), "task is not running: {:?}", curr.state());
         assert!(!curr.is_idle());
         if curr.is_init() {
-            EXITED_TASKS.lock().clear();
+            EXITED_TASKS.with_current(|exited_tasks| exited_tasks.clear());
             axhal::misc::terminate();
         } else {
             curr.set_state(TaskState::Exited);
-            curr.notify_exit(exit_code, self);
-            EXITED_TASKS.lock().push_back(curr.clone());
-            WAIT_FOR_EXIT.notify_one_locked(false, self);
-            self.resched(false);
+
+            // Notify the joiner task.
+            curr.notify_exit(exit_code);
+
+            // Push current task to the `EXITED_TASKS` list, which will be consumed by the GC task.
+            EXITED_TASKS.with_current(|exited_tasks| exited_tasks.push_back(curr.clone()));
+            // Wake up the GC task to drop the exited tasks.
+            WAIT_FOR_EXIT.with_current(|wq| wq.notify_one(false));
+            // Schedule to next task.
+            self.inner.resched(false);
         }
         unreachable!("task exited!");
     }
 
-    pub fn block_current<F>(&mut self, wait_queue_push: F)
-    where
-        F: FnOnce(AxTaskRef),
-    {
+    pub fn blocked_resched(&mut self) {
```

Comment:
Why call it `blocked_resched`?

Reply:
Corresponding to `preempt_resched`, it is primarily called by the `wait_xxx` methods in `wait_queue.rs` to check whether the current task is in a `BLOCKED` state.